ipykernel/tests/test_message_spec.py

```diff
@@ -419,6 +423,14 @@ def test_kernel_info_request():
     validate_message(reply, 'kernel_info_reply', msg_id)
 
 
+def test_comm_info_request():
+    flush_channels()
+
+    msg_id = KC.comm_info()
```

Comment:
Can you add a `if not hasattr(KC, 'comm_info'): raise SkipTest()`? Then this test will run while comm_info is defined, but skip while jupyter_client is < 4.1.